tests/model_fields_/test_arrayfield.py

```diff
@@ -634,6 +634,21 @@ def test_overlap_values(self):
             self.objs[:3],
         )
 
+    def test_overlap_empty_values(self):
+        qs = NullableIntegerArrayModel.objects.filter(order__lt=-30)
+        self.assertCountEqual(
+            NullableIntegerArrayModel.objects.filter(
+                field__overlap=qs.values_list("field"),
+            ),
+            [],
+        )
+        self.assertCountEqual(
+            NullableIntegerArrayModel.objects.filter(
+                field__overlap=qs.values("field"),
+            ),
+            [],
+        )
```

Comment:
I believe this is additional test coverage, not regression tests. Additional test coverage be separate commits that precede refactoring to make it clear that a refactor does not change behavior.